### PR TITLE
feature/exception_middleware add exception middleware

### LINKED ISSUE #1576: Feature: add ExceptionMiddleware to handle processing exceptions
We should provide users with a suitable mechanism to handle and process application exceptions like [**FastAPI** does](https://fastapi.tiangolo.com/reference/fastapi/#fastapi.FastAPI.exception_handler)

I think the best way to make such mechanism suitable and flexable in the same time - implement it via special built-in middleware like in the following example:

Register handlers by constuctor:

```python
from faststream import ExceptionMiddleware
from faststream.nats import NatsBroker

async def handle_value_exception(exc: ValueError): ...

async def handle_type_exception(exc: TypeError): ...

exception_handle_middleware = ExceptionMiddleware(
    {
        ValueError: handle_value_exception,
        TypeError: handle_type_exception,
    }
)

broker = NatsBroker(middlewares=(exception_handle_middleware,))
```

And register handlers by decorator:

```python
from faststream import ExceptionMiddleware
from faststream.nats import NatsBroker

exception_handle_middleware = ExceptionMiddleware()

@exception_handle_middleware.add_handler(ValueError)
async def handle_value_exception(exc: ValueError): ...

@exception_handle_middleware.add_handler(TypeError)
async def handle_type_exception(exc: TypeError): ...

broker = NatsBroker(middlewares=(exception_handle_middleware,))
```

### LINKED ISSUE #1617: RequestValidationError handler when using NATS and FastAPI
I'm not sure if this is a bug or just a usage issue but I can't handle RequestValidationError when messages are invalid.

I tried the FastAPI approach as documented [here](https://fastapi.tiangolo.com/tutorial/handling-errors/#override-request-validation-exceptions) but the handler is never called even though RequestValidationError exception is raised and not handled except by FastAPI's logger, in our case.

Please assist.

### PR DESCRIPTION
**Pull request by Rusich90**
fixes #1576, fixes #1617

Draft: Need create test and check linters

You can use ExceptionMiddleware like examples in issue

```python
from pydantic import ValidationError

from faststream.broker.middlewares.exception import ExceptionMiddleware
from faststream.kafka import KafkaBroker

exception_middleware = ExceptionMiddleware()


@exception_middleware.add_handler(ZeroDivisionError)
async def handle_type_exception(exc: ZeroDivisionError): ...


@exception_middleware.add_handler(ValidationError)
async def handle_validation_error(exc: ValidationError): ...


broker = KafkaBroker("localhost:29092", middlewares=[exception_middleware])

```

or like this

```python
from pydantic import ValidationError

from faststream.broker.middlewares.exception import ExceptionMiddleware
from faststream.kafka import KafkaBroker


async def handle_type_exception(exc: ZeroDivisionError): ...


async def handle_validation_error(exc: ValidationError): ...


exception_handlers = {
    ZeroDivisionError: handle_type_exception,
    ValidationError: handle_validation_error,
}

exception_middleware = ExceptionMiddleware(
    exception_handlers=exception_handlers
)

broker = KafkaBroker("localhost:29092", middlewares=[exception_middleware])

```

### CODE REVIEW

File: faststream/broker/middlewares/exception.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        self,
+        exception_handlers: Optional[
+            Dict[Exception, Callable[[Exception], "SendableMessage"]]
+        ] = None
+    ) -> None:
+        if not exception_handlers:
+            exception_handlers = {}
+        self._exception_handlers = exception_handlers
+
+    def add_handler(self, exc: Exception) -> Callable[[Exception], None]:
```
Comment (Lancetnik) [inline faststream/broker/middlewares/exception.py:50]: Should return `Callable[[Callable[[Exception], SendableMessage], None]`

**Review by Lancetnik — CHANGES_REQUESTED** [review]
Test is too large, can u split it for specific ones?

**Comment by sheldygg** [comment]
I made some improvements https://github.com/Rusich90/faststream_fork/pull/1

**Review by Lancetnik — APPROVED** [review]

**Comment by Lancetnik** [comment]
@kumaranvpl can you, please, check the documentation and merge this one?

**Comment by kumaranvpl** [comment]
> @kumaranvpl can you, please, check the documentation and merge this one?

Proofread the docs. Will merge it now.

**Review by kumaranvpl — APPROVED** [review]

**Pull request merged**